enterprise/backend/src/metabase_enterprise/serialization/api.clj

```diff
@@ -0,0 +1,237 @@
+      (close [_]
+        (let [^AbstractConfiguration config (logger/configuration)]
+          (.removeLogger config (.getName logger))
+          (.stop appender)
+          ;; this method is only present in AbstractConfiguration
+          (.removeAppender config (.getName appender))
+          (.updateLoggers (logger/context)))))))
+
+;;; Logic
+
+(defn- serialize&pack ^File [opts]
+  ;; TODO: how to get time in local timezone?
+  (let [id       (format "%s-%s"
+                         (u/slugify (public-settings/site-name))
+                         (u.date/format "YYYY-MM-dd_HH-mm" (t/zoned-date-time)))
+        path     (io/file parent-dir id)
+        dst      (io/file (str (.getPath path) ".tar.gz"))
+        log-file (io/file path "export.log")]
+    (with-open [_logger (make-logger 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (-> (extract/extract opts)
+            (storage/store! path))
+        ;; not removing storage immediately to save some time before response
+        (compress-tgz path dst)
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:archive  (when (.exists dst)
+                 dst)
+     :log-file (when (.exists log-file)
+                 log-file)
+     :callback (fn []
+                 (when (.exists path)
+                   (run! io/delete-file (reverse (file-seq path))))
+                 (when (.exists dst)
+                   (io/delete-file dst)))}))
+
+(defn- unpack&import [^File file & [size]]
+  (let [dst      (io/file parent-dir (apply str (repeatedly 20 #(char (+ 65 (rand-int 26))))))
+        log-file (io/file dst "import.log")]
+    (with-open [_logger (make-logger 'metabase-enterprise.serialization log-file)]
+      (try                              ; try/catch inside logging to log errors
+        (log/infof "Serdes import, size %s" size)
+        (let [path (uncompress-tgz file dst)]
+          (serdes/with-cache
+            (-> (v2.ingest/ingest-yaml (.getPath (io/file dst path)))
+                (v2.load/load-metabase! {:abort-on-error true}))))
+        (catch Exception e
+          (log/error e "Error during serialization"))))
+    {:log-file log-file
+     :callback #(when (.exists dst)
+                  (run! io/delete-file (reverse (file-seq dst))))}))
+
+(comment
+  ;; dump on disk and see that it works
+  (serialize&pack {:no-data-model true :no-settings true :targets [["Collection" 31409]]})
+  (unpack&import (io/file "metabase_test.tar.gz")))
+
+;;; HTTP API
+
+(api/defendpoint POST "/export"
+  "Serialize and retrieve Metabase data."
+  [:as {{:strs [all-collections collection settings data-model field-values database-secrets]
+         :or   {settings        true
+                data-model      true
+                all-collections true}}
+        :query-params}]
+  {collection       [:maybe [:or ms/PositiveInt [:sequential ms/PositiveInt]]]
```

Comment:
```suggestion
  {collection_ids       [:maybe [:or ms/PositiveInt [:sequential ms/PositiveInt]]]
```

Reply:
it's going to look like `collection_ids=1&collection_ids=2`. To be in line with how query params are generated usually we could go with `collection-id=1&collection-id=2` then, but I don't see how that is an improvement.

Reply:
> but I don't see how that is an improvement

Why not tho? IMO, the more specific, the better here since it's a public API.